src/contracts/core/DelegationManager.sol

```diff
@@ -785,6 +1018,146 @@ contract DelegationManager is
         }
     }
 
+    /**
+     * @notice Sets the withdrawal delay (in seconds) for each strategy in `_strategies` to `_withdrawalDelay`.
+     * gets called when initializing contract or by calling `setStrategyWithdrawalDelay`
+     */
+    function _setStrategyWithdrawalDelay(
+        IStrategy[] calldata _strategies,
+        uint256[] calldata _withdrawalDelays
+    ) internal {
+        require(
+            _strategies.length == _withdrawalDelays.length,
+            "DelegationManager._setStrategyWithdrawalDelay: input length mismatch"
+        );
+        uint256 numStrats = _strategies.length;
+        for (uint256 i = 0; i < numStrats; ++i) {
+            IStrategy strategy = _strategies[i];
+            uint256 prevStrategyWithdrawalDelay = strategyWithdrawalDelays[strategy];
+            uint256 newStrategyWithdrawalDelay = _withdrawalDelays[i];
+            require(
+                newStrategyWithdrawalDelay <= MAX_WITHDRAWAL_DELAY,
+                "DelegationManager._setStrategyWithdrawalDelay: _withdrawalDelay cannot be > MAX_WITHDRAWAL_DELAY"
+            );
+
+            // set the new withdrawal delay (in seconds)
+            strategyWithdrawalDelays[strategy] = newStrategyWithdrawalDelay;
+            emit StrategyWithdrawalDelaySet(strategy, prevStrategyWithdrawalDelay, newStrategyWithdrawalDelay);
+        }
+    }
+
+    /**
+     *
+     *                         SLASHING AND SHARES HELPER FUNCTIONS
+     *
+     */
+
+    /**
+     * @notice helper pure to calculate the scaledShares to store in queue withdrawal
+     * and the shares to decrement from StrategyManager/EigenPodManager
+     */
+    function _calculateSharesToQueueWithdraw(
+        uint256 sharesToWithdraw,
+        uint256 stakerScalingFactor,
+        uint64 totalMagnitude
+    ) internal pure returns (uint256 sharesToDecrement, uint256 scaledStakerShares) {
+        // TODO: DOUBLE CHECK THIS BEHAVIOR
+        // NOTE that to prevent numerator overflow, the max sharesToWithdraw is
+        // x*1e36 <= 2^256-1
+        // => x <= 1.1579e41
+        sharesToDecrement =
+            (sharesToWithdraw * SlashingConstants.PRECISION_FACTOR_SQUARED) / (stakerScalingFactor * totalMagnitude);
+        scaledStakerShares = sharesToWithdraw * stakerScalingFactor / SlashingConstants.PRECISION_FACTOR;
```

Comment:
can u switch this to the logic we discussed instead?

Reply:
also `scaledStakerShares` should be `sharesToWithdraw * INITIAL_TOTAL_MAGNITUDE / operatorTotalMagnitude`.

I had a mistake in my doc. Lmk if you want to chat about this